File: lib/feature/pose/pose_channel.dart
import 'package:shus_flutter/common/abstracts/failure.dart';
import 'package:shus_flutter/common/abstracts/result.dart';
import 'package:shus_flutter/common/abstracts/success.dart';
import 'package:shus_flutter/common/interface/native_channel.dart';
import 'package:shus_flutter/feature/pose/result/pose_error.dart';
import 'package:shus_flutter/feature/pose/pose_input.dart';
import 'package:shus_flutter/feature/pose/result/pose_output.dart';

class PoseChannel extends NativeChannel {
  Future<Result<PoseOutput, PoseError>> poseDetect(PoseInput input) async {
    final result = await channel.invokeMethod("pose-fn", input.toData());
    final mapData = Map<String, dynamic>.from(result);
    bool isSuccess = mapData["result"];
    if (isSuccess) {
      return Success(PoseOutput.fromData(mapData));
    } else {
      return Failure(PoseError.fromData(mapData));
    }
  }
}

File: android/app/src/main/kotlin/com/example/shus_flutter/feature/pose/PoseChannel.kt
package com.example.shus_flutter.feature.pose

import android.graphics.Bitmap
import com.github.michaelbull.result.*
import com.google.mlkit.vision.pose.PoseDetection
import com.google.mlkit.vision.pose.PoseDetector
import com.google.mlkit.vision.pose.accurate.AccuratePoseDetectorOptions
import java.util.*

class PoseChannel {

    var poseDetector: PoseDetector

    init {
        val detectorOptions = AccuratePoseDetectorOptions.Builder()
            .setDetectorMode(AccuratePoseDetectorOptions.SINGLE_IMAGE_MODE)
            .build()
        poseDetector = PoseDetection.getClient(detectorOptions)
    }

    fun poseDetect(args: HashMap<String, Any>?, completion: (Result<PoseSuccess, PoseError>) -> Unit) {
        val input = PoseInput(args)
        val flutterImage = input.image

        if (flutterImage == null) {
            val error = PoseError.IMAGENOTFOUND
            completion(Err(error))
        } else {
            val inputImage = flutterImage.toInputImage()
            val renderedImage = flutterImage.bitmap.copy(Bitmap.Config.ARGB_8888, true)
            poseDetector.process(inputImage)
                .addOnSuccessListener { pose ->
                    val success = PoseSuccess(pose, renderedImage)
                    completion(Ok(success))
                }
                .addOnFailureListener {
                    val error = PoseError.DETECT
                    completion(Err(error))
                }
        }
    }
}